File: src/main/java/com/back/domain/member/memberGame/dto/MemberGameDto.java
package com.back.domain.member.memberGame.dto;

import com.back.domain.member.memberGame.StatusEnum;
import com.back.domain.member.memberGame.entity.MemberGame;

public record MemberGameDto(
        int id,
        String platform,
        double playtime,
        boolean isFavorite,
        StatusEnum status,
        Long gameId,
        Long igdbId,
        String gameName,
        String coverImageId,
        Integer reviewId,
        Double rating
){
    public MemberGameDto(MemberGame memberGame){
        this(
                memberGame.getId(),
                memberGame.getPlatformGroupName(),
                memberGame.getPlaytime(),
                memberGame.isFavorite(),
                memberGame.getStatus(),
                memberGame.getGame().getId(),
                memberGame.getGame().getIgdbId(),
                memberGame.getGame().getName(),
                memberGame.getGame().getCoverImageId(),
                memberGame.getReview() != null ? memberGame.getReview().getId() : null,
                memberGame.getReview() != null ? memberGame.getReview().getRating() : null
        );
    }
}

File: src/main/kotlin/com/back/domain/member/memberGame/dto/MemberGameDto.kt
package com.back.domain.member.memberGame.dto

import com.back.domain.member.memberGame.StatusEnum
import com.back.domain.member.memberGame.entity.MemberGame
import com.fasterxml.jackson.annotation.JsonProperty

data class MemberGameDto(
    val id: Int,
    val platform: String?,
    val playtime: Double,
    @get:JsonProperty("isFavorite") val isFavorite: Boolean,
    val status: StatusEnum?,
    val gameId: Long?,
    val igdbId: Long?,
    val gameName: String?,
    val coverImageId: String?,
    val reviewId: Int?,
    val rating: Double?,
) {
    constructor(memberGame: MemberGame) : this(
        id = memberGame.id,
        platform = memberGame.platformGroupName,
        playtime = memberGame.playtime,
        isFavorite = memberGame.isFavorite,
        status = memberGame.status,
        gameId = memberGame.game.id,
        igdbId = memberGame.game.igdbId,
        gameName = memberGame.game.name,
        coverImageId = memberGame.game.coverImageId,
        reviewId = memberGame.review?.id,
        rating = memberGame.review?.rating,
    )
}
